### ISSUE TITLE
Admin-FE SSO

### ISSUE THREAD

**Issue by deltatux** [BDX-town/Mangane#115]
Hi Mangane!

Loving this FE, one of the things I miss from Pleroma-FE is the ability to click a button that will open up Admin-FE and automatically log into Admin-FE via SSO.

Was wondering if we can have a similar feature in Mangane to make instance management easier? It also doesn't help that Admin-FE doesn't support any accounts with OTP enabled.

Thanks!

**Comment by Cl0v1s** [comment]
It will help me for administration on bdx.town of course too ! 
I did not know about that pleroma-fe feature, do you have any documentation about it ? :)

**Comment by bugsysop** [comment]
I have access to Admin FE with this menu: « Ouvrir l’interface de...» (third from Bottom) accessible from profile page. Also the fact that the option name is truncated dosn’t help.

![Capture d’écran 2022-12-25 à 06 40 16](https://user-images.githubusercontent.com/365281/209458152-691a7e72-a159-405d-9e1d-912d2302fe2b.png)

**Comment by bugsysop** [comment]
@deltatux Can you be more explicit about this:

`Admin-FE doesn't support any accounts with OTP enabled`

**Comment by deltatux** [comment]
> It would help me with administration on bdx.town too ! I did not know about that pleroma-fe feature, do you have any documentation about it ? :)

So I went through the Pleroma docs and I can't seem to find any mention of this SSO mechanism. Could just be baked into the code. Unfortunately, I'm still new to Akkoma/Pleroma & I'm not a dev, so not sure where in the code to look to see where this was implemented.

If I run into it, I'll be sure to link it to this ticket.

> @deltatux Can you be more explicit about this:
> 
> `Admin-FE doesn't support any accounts with OTP enabled`

Just realized that in Akkoma/Pleroma, OTP may mean something else. What I mean is if you have multifactor authentication enabled on an admin account, it will just error out saying MFA required but doesn't do the MFA challenge Using the SSO login method where Pleroma-FE handles the login and passes the login token to Admin-FE allows you login into an account with OTP enabled.

Here's an example of the error you see trying to login to Admin-FE on an account with MFA enabled:
![image](https://user-images.githubusercontent.com/59618028/209479619-91ca5009-9693-43ff-8ab8-ce6b85baa38a.png)

**Comment by Cl0v1s** [comment]
To understand well, when you want to log in `admin-fe` with your account using 2FA, you get this error `mfa_required` ?

**Comment by deltatux** [comment]
> To understand well, when you want to log in `admin-fe` with your account using 2FA, you get this error `mfa_required` ?

Yes that is correct and the only way to get around this has been either:

- Login via Pleroma-FE and use its SSO to log into Admin-FE, the 2FA challenge is performed by Pleroma-FE
- Create a second admin user without 2FA enabled to log into Admin-FE directly.

**Comment by thomasjwebb** [comment]
>     * Login via Pleroma-FE and use its SSO to log into Admin-FE, the 2FA challenge is performed by Pleroma-FE

How do you do this when Mangane is enabled?
